RELEASE.md

```diff
@@ -0,0 +1,52 @@
+# Release Process for `connect-go`
+
+## Preparing for Release
+
+1. **Update Version Number**
+   - In `connect.go`, modify the `Version` constant to match the target release according to [semver](https://semver.org/) standards. Remove the `-dev` suffix.
+     ```go
+     const Version = "X.Y.Z"
+     ```
+
+2. **Version Requirements (Optional)**
+   - If changes require a new API in `connect-go`, declare a constant referencing the updated code.
+     ```go
+     const (
+         IsAtLeastVersionX_Y_Z = true
+     )
+     ```
+   - Ensure the generated code refers to this constant. This is used as a compile time check of source compatibilty.
```

Comment:
The Notion document is more specific about how to do this (i.e. specific file to edit), and we should keep that level of specificity in the README. A sample diff/PR could be included here too.